stable_baselines3/her/her_replay_buffer.py

```diff
@@ -325,22 +325,24 @@ def _sample_goals(self, batch_inds: np.ndarray, env_indices: np.ndarray) -> np.n
 
         if self.goal_selection_strategy == GoalSelectionStrategy.FINAL:
             # replay with final state of current episode
-            goal_ep_inds = batch_ep_length - 1
+            transition_indices_in_episode = batch_ep_length - 1
 
         elif self.goal_selection_strategy == GoalSelectionStrategy.FUTURE:
             # replay with random state which comes from the same episode and was observed after current transition
-            batch_idx_within_ep = batch_inds - batch_ep_start
-            goal_ep_inds = np.random.randint(batch_idx_within_ep, batch_ep_length)
+            current_indices_in_episode = batch_inds - batch_ep_start
+            # Note(antonin): current implementation does not seem correct, a +1
```

Comment:
> Should we strictly stick to the strictly future strategy?

Yes, otherwise we are not following the paper/original implementation (I expect it to have some impact on performance even if it is a pain to handle special cases)

Reply:
👍

Reply:
I like the current version without max_episode_length from a research perspective, if performance trade off isn't too bad. Enforcing max_episode_length can make the env no longer a MDP and cause issues for off-policy algorithms, especially when `max_episode_length` is set to something small. A detailed explanation can be seen [here](https://github.com/rlworkgroup/metaworld/issues/192#issuecomment-752284550).

Reply:
see https://github.com/DLR-RM/stable-baselines3/issues/284
for how we handle infinite horizon problem with off policy algos.